in_toto/in_toto_sign.py

```diff
@@ -211,24 +258,46 @@ def main():
   # Override defaults in settings.py with environment variables and RCfiles
   in_toto.user_settings.set_settings()
 
+
+  # Additional argparse sanitization
+  # NOTE: This tool is starting to have many inter-dependent argument
+  # restrictions. Maybe we should make it less sophisticated at some point.
   if args.verify and (args.append or args.output):
     parser.print_help()
     parser.exit(2, "conflicting arguments: don't specify any of"
         " 'append' or 'output' when verifying signatures")
 
+  # Regular signing and GPG signing are mutually exclusive
+  if (args.key == None) == (args.gpg == None):
+    parser.print_help()
+    parser.exit(2, "wrong arguments: specify either `--key PATH [PATH ...]`"
+      " or `--gpg [KEYID [KEYID ...]]`")
+
+  # For gpg verification we must specify a keyid (no default key is loaded)
+  if args.verify and args.gpg != None and len(args.gpg) < 1:
+    parser.print_help()
+    parser.exit(2, "missing arguments: specify at least one keyid for GPG"
+      " signature verification (`--gpg KEYID ...`)")
+
   metadata = _load_metadata(args.file)
 
+  # Specific command line argument restrictions if we deal with links
   if metadata._type == "link":
-    if len(args.key) > 1:
+    # Above we check that it's either `--key ...` or `--gpg ...`
+    # Here we check that it is not more than one in each case when dealing
+    # with links
+    if ((args.key != None and len(args.key) > 1) or
+        (args.gpg != None and len(args.gpg) > 1)):
       parser.print_help()
-      parser.exit(2, "wrong arguments: Link metadata can only be signed by"
-          " one key")
+      parser.exit(2, "too many arguments: Link metadata can not be signed by"
+          " multiple keys")
```

Comment:
Why can't multiple keys sign for link metadata?  Not fully supported yet?  Limitation?  Prohibited?  Please explain in the exception message.

Reply:
Addressed in 778d10690dba95d67886f9a546a8778e1342ae30